src/blueapi/service/interface.py

```diff
@@ -117,27 +121,34 @@ def _update_scan_num(md: dict[str, Any]) -> int:
 
 def setup(config: ApplicationConfig) -> None:
     """Creates and starts a worker with supplied config"""
-
     set_config(config)
-
     set_up_logging(config.logging)
 
     # Eagerly initialize worker and messaging connection
-
     worker()
     stomp_client()
     if numtracker_client() is not None:
         context().run_engine.scan_id_source = _update_scan_num
-    _hook_run_engine_and_path_provider()
+        _hook_run_engine_and_path_provider()
 
 
-# TODO: https://github.com/DiamondLightSource/blueapi/issues/875
 def _hook_run_engine_and_path_provider() -> None:
-    path_provider = get_path_provider()
+    try:
+        path_provider = get_path_provider()
+    except NameError:
+        path_provider = None
+
     run_engine = context().run_engine
 
-    if isinstance(path_provider, StartDocumentPathProvider):
+    if path_provider is None:
+        path_provider = StartDocumentPathProvider()
+        set_path_provider(path_provider)
         run_engine.subscribe(path_provider.update_run, "start")
+    else:
+        logging.debug(
+            "A StartDocumentPathProvider has not been configured for numtracker"
+            f"because a different path provider was already set: {path_provider}"
+        )
```

Comment:
I may be contradicting my past self, but is this a valid use case? Would we ever expect someone to have numtracker config and also their own `PathProvider`? If not I think we should just raise

Reply:
No I don't think so. I'll make it raise